liblangutil/SourceReferenceExtractor.h

```diff
@@ -55,12 +55,12 @@ namespace SourceReferenceExtractor
 	struct Message
 	{
 		SourceReference primary;
-		std::string severity; // "Error", "Warning", "Info", ...
+		Error::Type type;
```

Comment:
I think this was actually meant to be just severity not the whole type:
```suggestion
		Error::Severity severity;
```
I.e. `Error` carries complete info about the error, including the type, but `Message` is just some subset of its info.

Reply:
I'm not sure this should be done. 

Earlier, `severity` was a string value that was freely abused. Several values of `severity` were being passed. These were not all provided in `Error::Severity` enum. 

So, I converted this to `Error::Type`. This provides for a bigger coverage for the error types. 

Otherwise, there are cases right now, where `error types` are printed ( they were passed as strings before . So, even though the print function was supposed to accept `severity`, it compiled anyway ) . Those will be reduced to one of `Error::Severity` values. That will lead to some information loss.

As against now, where we are getting extra information in our error messages by printing the exact error type.

A workaround for this would be to expand `Error::Severity` the way we have expanded `Error::Type`

Reply:
ok, I guess we could keep it as `Error::Type` here as long as https://github.com/ethereum/solidity/pull/13162#discussion_r939127367 is changed. I thought that other place was removing all the extra info anyway and dumbing it down to `Error::Severity` but if not, we can keep it as is.

Reply:
If we go ahead with https://github.com/ethereum/solidity/pull/13162#discussion_r939127367, we get errors on the CI 

For eg. https://circleci.com/gh/ethereum/solidity/1131923?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link  


Here, the test apparently does expect a proper error Type. But since we've constrained it to only be from `Error::Severity` , it throws this error. 

Had made a PR to propagate this change to solc-js  ( https://github.com/ethereum/solc-js/pull/643) , but it was refused.

Reply:
Also, indeed, you're right. Doing https://github.com/ethereum/solidity/pull/13162#discussion_r939127367  effectively dumbs it down to `Error::Severity` . Will push changes to make it that way if we figure our what to do with the error on solc-js side.

Reply:
Can you track down the path of code that `SyntaxError` from solc-js goes through? I.e. I don't fully understand why it's turning into `Error` even though this is not on the CLI. Are we using `SourceReferenceFormatter` for more than just CLI?

I'd really prefer a solution that would keep the current output unchanged and only reshuffle things internally...

Reply:
Trying to track the code at solc-js
`solc-js/test/cli.ts : 62 `

To be fair, the error does seem to be a genuine mismatch.  The file name is `cli.ts` . We do want our CLI tests to only show `severity` and not the full `type`. 

I'll explore more and find the exact function calls that lead it to our code change in the main `solidity` repo

Reply:
> To be fair, the error does seem to be a genuine mismatch.  The file name is `cli.ts` . We do want our CLI tests to only show `severity` and not the full `type`. 

It's complicated :) solc-js uses the Standard JSON interface and its CLI is just a thin wrapper over that. So it naturally gets full error type from JSON. Maybe it should convert that to severity internally but I don't think it's that important. It's not 1:1 equivalent to solc anyway.

For now just assume that in solc-js we want to keep it as `SyntaxError`.

Reply:
Also, I see you still haven't changed this field back to `Error::Type`.

Reply:
Okay. So, how are we supposed to go ahead here ? 

Because, at the moment, it's getting 'Error:...` from solc and it expects `SyntaxError` . I'm not sure where to make the change

Reply:
I tried to reproduce CI [ems_solcjs failing test ](https://app.circleci.com/pipelines/github/ethereum/solidity/25722/workflows/ad2d4a51-1a91-4884-9a0e-730ad5a83d42/jobs/1135117) locally, but I was not able to... any clues why this should be? Locally the node and npm versions are different from CI... 
Not that I think that node/npm versions are the problem here, but I got curious about this difference in behaviour when testing local and maybe this can help identify a solution here.

Reply:
Probably because it was my proposal to change it to `Error::Severity` that was responsible for it. Now that it's back to `Error::Type`, there should be no error.